### DISCUSSION TITLE
In version 4.18.1, under the VPC network resource domain, DHCP dynamic IP addresses cannot be allocated across hosts

### DISCUSSION THREAD

**Discussion by Qinshan886** [apache/cloudstack#9320] [Q&A]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
Virtual Router
~~~

##### CLOUDSTACK VERSION

~~~
4.18.1
~~~

##### OS / ENVIRONMENT

Centos 7

##### SUMMARY

I am using CloudStack 4.18.1. After creating a resource domain of the VPC network type, I added two host nodes.
![image](https://github.com/apache/cloudstack/assets/173532830/7de61159-7bd8-42e3-8b2e-1d039895ca38)
And created a VPC network (it automatically creates a virtual router on host node 1) and added the network layer
![image](https://github.com/apache/cloudstack/assets/173532830/698dd1c1-a515-4bba-8b33-c0d3b781efb4)
![image](https://github.com/apache/cloudstack/assets/173532830/cb7bcbb2-e940-4433-91f4-fe71ba358961)

##### STEPS TO REPRODUCE

When I started to create the first instance machine, I chose the hk-node1 node. It is on the same host node as the vpc virtual router
![image](https://github.com/apache/cloudstack/assets/173532830/a4a5a7a8-59c2-42a9-bba8-8a1619a705ae)

The instance machine and the virtual router can be connected normally, and the instance machine can request the DHCP of the virtual router of the vpc to obtain the IP address
![image](https://github.com/apache/cloudstack/assets/173532830/b391b128-71c9-427b-b866-b556500aba4f)
The above picture shows the log information of DHCP in the vpc virtual router

![image](https://github.com/apache/cloudstack/assets/173532830/be487001-4fc4-4b75-ad89-639ed5819a20)
![image](https://github.com/apache/cloudstack/assets/173532830/cfc781ae-c44d-4ec1-922e-f226ba764a44)
The above picture shows the log and IP information of the instance machine requesting the vpc virtual router during the dhcp process.this is normal

==================================hk-node2 host node dividing line============================
I now create an instance machine on the hk-node2 host node
![image](https://github.com/apache/cloudstack/assets/173532830/302c5fee-52c0-4190-88c6-89477df22ec9)

![image](https://github.com/apache/cloudstack/assets/173532830/515144ee-601c-4bbe-8df8-a8ba0f34f416)
The above picture shows the vpc virtual router log (it is on the hk-node1 host node), and there is no log showing the IP address assigned to the instance machine on the hk-node2 host node.

![image](https://github.com/apache/cloudstack/assets/173532830/d2849701-61ff-4b03-b80e-b1a4e02090d0)
![image](https://github.com/apache/cloudstack/assets/173532830/00dea9c1-1ef1-42c7-b005-a6fab634119c)
The above figure shows that the instance machine on the hk-node2 host node cannot access the gateway address of the vpc virtual router 10.1.100.1 during the dhcp request process, resulting in a dhcp timeout. Failed to obtain the IP address；

This is abnormal. Why can't the instance machine on the hk-node2 host node obtain an IP address when the two host nodes are in the same resource domain cluster? I tried to reproduce this operation in the latest version 4.19, and the problem of not being able to obtain an IP address across host nodes also exists. Please help!!!

##### EXPECTED RESULTS
The instance machine I created on the hk-node2 host node should be able to obtain the IP address dynamically allocated by DHCP normally, just like the instance machine on the hk-node1 host node.

##### ACTUAL RESULTS
The actual result is that the instance machine I created on another host node cannot connect to the host node where the vpc virtual router is located, resulting in the instance machine on other host nodes being unable to dynamically obtain an IP address through dhcp. In simple terms, the instance machine I created must be on the same host node as the vpc virtual router. If it is not on the same host node, the instance machine cannot obtain an IP

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by rajujith** [comment]
@Qinshan886  could you check the VPC tier VLAN is trunked on both host's switch ports on the physical switch?

**Comment by Qinshan886** [comment]
@rajujith Sorry, I don't know much about this in terms of networking. I don't know how to check it; but my two host nodes are virtual machines, not physical servers. The network deployment also uses the bridge described in the official manual

**Comment by rajujith** [comment]
@Qinshan886  So you have added Nested KVM hosts in the cloudstack. What is your base hypervisor? Could you check whether the vNICs of the nested KVM that are associated with the cloudstack guest traffic are in the same bridge/vSwitch? and also allows the tagged VLAN used in the VPC tier.

**Comment by Qinshan886** [comment]
@rajujith My base hypervisor is kvm and libvirtd. The NIC for guest traffic is on the same bridge。It also allows the use of tagged VLANs at the VPC level.

**Comment by Qinshan886** [comment]
@rajujith I have also tried to deploy and build according to the official website documentation on the latest version 4.19 and found that the vpc cannot dynamically allocate IP addresses across hosts using dhcp. Has anyone else encountered this problem?

**Comment by DaanHoogland** [comment]
closed this for lack of interaction, please reopen or open a new item if applicable
